### PR TITLE
Add cosmology, gaussiancl, transformcl, and glass

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
@conda-forge/help-python-c, ready for review!

File: recipes/cosmology/meta.yaml
Comment (xhochy) [inline recipes/cosmology/meta.yaml:3]:
```suggestion
```
Only set this if you need 3.10+

File: recipes/gaussiancl/meta.yaml
Comment (xhochy) [inline recipes/gaussiancl/meta.yaml:3]:
```suggestion
```

File: recipes/glass/meta.yaml
Comment (xhochy) [inline recipes/glass/meta.yaml:13]: You can make this noarch. The CI on Windows will fail with a not-sovable error but we can still merge

File: recipes/transformcl/meta.yaml
Comment (xhochy) [inline recipes/transformcl/meta.yaml:3]:
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by Saransh-cpp** [comment]
Thanks for the review, @xhochy! Should be ready again!

**Comment by Saransh-cpp** [comment]
The CI is erroring out with -
```
conda_build.exceptions.CondaBuildUserError: Failed to render jinja template in /home/conda/staged-recipes-copy/recipes/cosmology/meta.yaml:
'python_min' is undefined
```

so I guess, I need to set the variable?

**Comment by xhochy** [comment]
No, this is a CI issue in staged-recipes. Somehow it doesn't work at the moment if not all recipes are `noarch: python`. Can you submit the other recipe as an individual PR, please?

**Comment by Saransh-cpp** [comment]
Ah, I see, thank you! I have created a new PR for `flt`.

Blocked by #28411

**Comment by Saransh-cpp** [comment]
Ahh, same error :(

Should I replace `python_min` with the actual version in `requirements`?

**Comment by Saransh-cpp** [comment]
Not using `{% set python_min = "3.9" %}` gives me an error in the CI, and not using `python_min` altogether throws a linter warning. I have added `python_min` and `{% set python_min = "3.9" %}` back, but please let me know if I am doing something wrong!

**Comment by Saransh-cpp** [comment]
Hi @matthewfeickert, thanks for the review! I was talking about https://github.com/conda-forge/staged-recipes/pull/28405#issuecomment-2508541328 in the comment related to the CI failure above.

> which I've been told by the conda-forge/core team before is "fine" given

Yes, this should be fine. I was worried about the `python_min` variable, which @xhochy suggested [that I should only set if I need Python 3.10+](https://github.com/conda-forge/staged-recipes/pull/28405#discussion_r1863739052).

> Though as mentioned, probably better to submit each project as its own standalone PR in the future

`flt` (`python-c` package) is already in a separate PR (merged!) - #28411 - as suggested above, and all of these packages are pure Python.

I have squashed the commits so this should be ready for a review again.

@conda-forge/help-python, ready for review!

**Comment by matthewfeickert** [comment]
> Yes, this should be fine. I was worried about the `python_min` variable, which @xhochy suggested [that I should only set if I need Python 3.10+](https://github.com/conda-forge/staged-recipes/pull/28405#discussion_r1863739052).

Ah, yeah, while that (enforcing lower bounds that were more aggressive than the current subset of CPython versions conda-forge enforces) was the original intended purpose of `python_min` (AFAIK), I added an example to https://deploy-preview-2370--conda-forge-previews.netlify.app/docs/maintainer/knowledge_base/#noarch-python in https://github.com/conda-forge/conda-forge.github.io/pull/2370 of why `python_min` is preferable for every `noarch: python` feedstock as it actively enforces the `requires-python` metadata for the Python package.

**Comment by Saransh-cpp** [comment]
Ah I see, so I am guessing it is also required to `set` the `python_min` variable at top for the workflows to work? (which is how it is in this PR, so it should be ready to review)

**Comment by matthewfeickert** [comment]
> Ah I see, so I am guessing it is also required to `set` the `python_min` variable at top for the workflows to work?

Yeah, either a Jinja2 variable as you describe, or as a map in `recipe/conda_build_config.yaml` (c.f. https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python). The Jinja2 variable is easier though as you can keep everything in the `meta.yaml`.

**Comment by Saransh-cpp** [comment]
Thank you! This should be ready then ✅

**Comment by Saransh-cpp** [comment]
> No, this is a CI issue in staged-recipes. Somehow it doesn't work at the moment if not all recipes are noarch: python. Can you submit the other recipe as an individual PR, please?

I found the issue. The CI errors out with `python_min` not found if there are multiple packages in the same PR. It does not matter if they are `noarch: python` or not.

I will raise this as an issue. Closing this PR as I am splitting it into multiple individual package PRs.

**Pull request closed without merging**
